trustfall_core/src/frontend/mod.rs

```diff
@@ -1114,7 +1113,7 @@ where
                         .or_default()
                         .push(subfield_name.clone());
 
-                    (subfield_name, subfield_raw_type, SmallVec::from([subfield]))
+                    (subfield_name, subfield_raw_type.clone(), SmallVec::from([subfield]))
```

Comment:
We can get a much cheaper clone if we place this in an Rc, infact, we can do this everywhere if we keep `Type` immutable.